android/src/main/java/com/raulh82vlc/MoviesFinder/ui/activities/MovieDetailsActivity.java

```diff
@@ -190,13 +191,20 @@ protected void setInitialTitle() {
 
     @Override
     protected void setToolbarInitialisation() {
-        mToolbar.setNavigationIcon(R.drawable.ic_back);
+        //mToolbar.setNavigationIcon(R.drawable.ic_back);
         setSupportActionBar(mToolbar);
-        final ActionBar actionBar = getSupportActionBar();
 
-        if (actionBar != null) {
-            actionBar.setDisplayHomeAsUpEnabled(true);
+        if (getSupportActionBar() != null) {
```

Comment:
you are calling to `getSupportActionBar()` until 3 times, which is a bad thing in terms of performance, I highly recommend you to maintain the previous implementation of the _line 195_ and using `actionBar` instead of this getter